### ISSUE TITLE
java.lang.RuntimeException

### ISSUE THREAD

**Issue by darkstarx** [Lyokone/flutterlocation#480]
**Describe the bug**
App crashes (in Android vitals)

**Tested on:**
 - Android 10, API Level 29, real devices (i.e. Poco F1, Poco F2 Pro)

**Additional logs**

``` shell
java.lang.RuntimeException: 
  at android.app.ActivityThread.performDestroyActivity (ActivityThread.java:5038)
  at android.app.ActivityThread.handleDestroyActivity (ActivityThread.java:5067)
  at android.app.servertransaction.DestroyActivityItem.execute (DestroyActivityItem.java:44)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2048)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:227)
  at android.app.ActivityThread.main (ActivityThread.java:7582)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:953)
Caused by: java.lang.NullPointerException: 
  at com.lyokone.location.FlutterLocationService.a (FlutterLocationService.java:2)
  at com.lyokone.location.LocationPlugin.c (LocationPlugin.java:51)
  at com.lyokone.location.LocationPlugin.detachActivity (LocationPlugin.java)
  at com.lyokone.location.LocationPlugin.access$000 (LocationPlugin.java)
  at com.lyokone.location.LocationPlugin.initialize (LocationPlugin.java)
  at com.lyokone.location.LocationPlugin.onAttachedToActivity (LocationPlugin.java)
  at com.lyokone.location.LocationPlugin.onAttachedToEngine (LocationPlugin.java)
  at com.lyokone.location.LocationPlugin.onDetachedFromActivity (LocationPlugin.java)
  at io.flutter.embedding.engine.FlutterEnginePluginRegistry.b (FlutterEnginePluginRegistry.java:52)
  at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach (FlutterActivityAndFragmentDelegate.java:58)
  at io.flutter.embedding.android.FlutterActivity.onDestroy (FlutterActivity.java:10)
  at android.app.Activity.performDestroy (Activity.java:8180)
  at android.app.Instrumentation.callActivityOnDestroy (Instrumentation.java:1342)
  at android.app.ActivityThread.performDestroyActivity (ActivityThread.java:5023)
  at android.app.ActivityThread.handleDestroyActivity (ActivityThread.java:5067)
  at android.app.servertransaction.DestroyActivityItem.execute (DestroyActivityItem.java:44)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2048)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:227)
  at android.app.ActivityThread.main (ActivityThread.java:7582)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:953)
```

**Comment by maciejpigulski** [comment]
From the stacktrace it seems it happens when an Activity is dismissed? What kind of dismiss scenario this is exactly?

**Comment by HiroshiHorie** [comment]
My issue may be related #479

**Comment by darkstarx** [comment]
> My issue may be related #479
Yes, it seems to be the same issue.

@maciejpigulski I can't to reply this crash manually, it's from Google Play Console. I think this crash could even stay invisible for user for the first time (with no intent to send crash report to Google).

Obviously, `LocationPlugin. onDetachedFromActivity` handler can be called unexpectedly by platform while attaching to activity is in progress, so it's recommended to make the code more safety by checking the existence of instances such as `activityBinding.getActivity()` and others. Of course, this will only mask the the underlying problem.. Sorry, idk what exactly can lead to such behavior under the hood ((

**Comment by maciejpigulski** [comment]
I am trying to reproduce it but still no luck. I have tried various scenarios where activity is destroyed and in a simple app that has only one screen I wasn't able to reproduce this. It is very weird that on the same stack there is an `onDetachedFromActivity` call directly to `onAttachedToEngine` and then to `onAttachedToActivity`.

Maybe from your apps perspective you have some idea what kind of screen transitions or app start/termination scenarios could potentially cause this?

**Comment by darkstarx** [comment]
I have reproduced the bug in the debug build on the real device. Replacing the main.dart code with the simplest test code with empty page doesn't eliminate the problem, still had been crashing. So, the dart code of app doesn't affect the problem.
Then I rolled back the main.dart, cleared the project:
* `flutter clean`
* removed `build` folder
* removed `.flutter-plugins`, `.flutter-plugins-dependencies`, `.packages`, `pubspec.lock`
then I rebuilt android project, reinstalled on the same device.. and after that the problem disappeared.
